kstreams/engine.py

```diff
@@ -59,18 +64,19 @@ async def send(
                 value, headers=headers, value_serializer_kwargs=value_serializer_kwargs
             )
 
+        kafka_headers = None
         if headers is not None:
-            headers = encode_headers(headers)
+            kafka_headers = encode_headers(headers)
```

Comment:
I would remove direct references to kafka if possible in the code outside `backends.Kafka`

Reply:
yes, it makes sense